fairlearn/reductions/exponentiated_gradient/_lagrangian.py

```diff
@@ -42,10 +42,10 @@ def eval_from_error_gamma(self, error, gamma, lambda_vec):
         L_high -- value of the Lagrangian under the best
         response of the lambda player
         """
-        lambda_signed = self.constraints.lambda_signed(lambda_vec)
+        lambda_projected = self.constraints.project_lambda(lambda_vec)
         if self.opt_lambda:
-            L = error + np.sum(lambda_vec * gamma) \
-                - self.eps * np.sum(lambda_signed.abs())
+            L = error + np.sum(lambda_projected * gamma) \
+                - self.eps * np.sum(lambda_projected)
         else:
             L = error + np.sum(lambda_vec * gamma) \
```

Comment:
Given that this first line is common, should it go outside the `if` statement?

Reply:
the first L is based on lambda_projected (both lines), the second L is based on lambda_vec (both lines).

Reply:
Sorry, missed that :-(